runtime/ms-rest/lib/webResource.js

```diff
@@ -293,4 +298,244 @@ WebResource.prototype.pipeInput = function(inputStream, destStream) {
+ *  - 'accept-language' by default is set to 'en-US'
+ *  - 'x-ms-client-request-id' by default is set to a new Guid. To not generate a guid for the request, please set options.disableClientRequestId to true
+ *
+ * @param {boolean} [options.disableClientRequestId] When set to true, instructs the client to not set 'x-ms-client-request-id' header to a new Guid().
+ *
+ * @param {object|string|boolean|array|number|null|undefined} [options.body] - The request body. It can be of any type. This method will JSON.stringify() the request body.
+ *
+ * @param {boolean} [options.disableJsonStringifyOnBody] - Indicates whether this method should JSON.stringify() the request body. Default value: false.
+ *
+ * @param {boolean} [options.bodyIsStream] - Indicates whether the request body is a stream (useful for file upload scenarios).
+ *
+ * @returns {object} WebResource Returns the prepared WebResource (HTTP Request) object that needs to be given to the request pipeline.
+ */
+WebResource.prototype.prepare = function prepare(options) {
+  if (options === null || options === undefined || typeof options !== 'object') {
+    throw new Error('options cannot be null or undefined and must be of type object')
+  }
+
+  if (options.method === null || options.method === undefined || typeof options.method.valueOf() !== 'string') {
+    throw new Error('options.method cannot be null or undefined and it must be of type string.');
+  }
+
+  if (options.url && options.pathTemplate) {
+    throw new Error('options.url and options.pathTemplate are mutually exclusive. Please provide either of them.')
+  }
+
+
+  if ((options.pathTemplate === null || options.pathTemplate === undefined || typeof options.pathTemplate.valueOf() !== 'string') && (options.url === null || options.url === undefined || typeof options.url.valueOf() !== 'string')) {
+    throw new Error('Please provide either options.pathTemplate or options.url. Currently none of them were provided.');
+  }
+
+  //set the url if it is provided.
+  if (options.url) {
+    if (typeof options.url !== 'string') {
+      throw new Error('options.url must be of type \'string\'.');
+    }
+    this.url = options.url;
+  }
+
+  //set the method
+  if (options.method) {
+    var validMethods = ['GET', 'PUT', 'HEAD', 'DELETE', 'OPTIONS', 'POST', 'PATCH'];
+    if (validMethods.indexOf(options.method.toUpperCase()) === -1) {
+      throw new Error('The provided method \'' + options.method + '\' is invalid. Supported HTTP methods are: ' + JSON.stringify(validMethods));
+    }
+  }
+  this.method = options.method.toUpperCase();
+
+  //construct the url if path template is provided
+  if (options.pathTemplate) {
+    if (typeof options.pathTemplate !== 'string') {
+      throw new Error('options.pathTemplate must be of type \'string\'.');
+    }
+    if (!options.baseUrl) {
+      options.baseUrl = 'https://management.azure.com';
```

Comment:
WebResource.js is available in ms-rest and not ms-rest-azure. It can be done, but it will be super complicated. To make it easy for us (azure users) to use the method, I have defaulted it to that specific value. Frankly speaking, the usage of this method will be very less outside the scope of azure.